core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -191,7 +193,7 @@ private void createZkNamespace() {
     ZkUtils zkUtilsForNamespaceCreation = ZkUtils.apply(
         zkConnForNamespaceCreation,
         zkSessionTimeoutMs, zkSessionTimeoutMs,
-        JaasUtils.isZkSecurityEnabled());
+        JaasUtils.isZkSecurityEnabled() && zKAclsEnabled);
```

Comment:
This is indeed the expected behavior. I believe it's possible that some users may want to turn ACLs off. The reason is because with ACLs enabled, the Schema Registry SASL Kerberos principal must be the same as Kafka. Otherwise, the Schema Registry can't create the `_schemas` topic. I expect some users will be ok with the SR having the same principal, and others won't. Let me know if you disagree.